packages/msgs-derive/tests/macro.rs

```diff
@@ -36,6 +40,53 @@ enum TestMsg2 {
     Specific4,
 }
 
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum ExecuteError {
+    #[error("default error")]
+    Default,
+}
+
+#[cw_serde]
+#[derive(EnsurePermissions)]
+pub enum TestMsg3 {
+    #[permission(Any, Proxy(gateway1))]
+    Any,
+    #[permission(Elevated, Proxy(gateway1))]
+    Proxy1,
+    #[permission(Specific(gateway1), Proxy(gateway1))]
+    Proxy2,
+    #[permission(Any, Proxy(gateway1))]
+    Proxy3,
+    #[permission(Any, Proxy(gateway1, gateway2, gateway3))]
```

Comment:
keep at least one of them as `Specific`, `Proxy2` only checks what happens when you have the same name for both roles